org.eclipse.m2e.pde/src/org/eclipse/m2e/pde/MavenJarFeatureModel.java

```diff
@@ -0,0 +1,119 @@
+package org.eclipse.m2e.pde;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.equinox.internal.p2.publisher.eclipse.FeatureParser;
+import org.eclipse.equinox.p2.publisher.eclipse.FeatureEntry;
+import org.eclipse.pde.internal.core.NLResourceHelper;
+import org.eclipse.pde.internal.core.feature.AbstractFeatureModel;
+import org.eclipse.pde.internal.core.feature.Feature;
+import org.eclipse.pde.internal.core.feature.FeatureChild;
+import org.eclipse.pde.internal.core.feature.FeaturePlugin;
+import org.eclipse.pde.internal.core.ifeature.IFeature;
+import org.eclipse.pde.internal.core.ifeature.IFeatureChild;
+import org.eclipse.pde.internal.core.ifeature.IFeatureInfo;
+import org.eclipse.pde.internal.core.ifeature.IFeaturePlugin;
+
+@SuppressWarnings("restriction")
+public class MavenJarFeatureModel extends AbstractFeatureModel {
```

Comment:
I wonder if we probably can reuse the default feature model of PDE here?

Reply:
You're right. With the feature.xml, we can delegate the whole task to PDE.
I've updated the code to reflect that. Though PDE doesn't seem to be able to access the file if contained within a jar, so right it's necessary to extract it first.